### PR TITLE
Add Blocks machines to ultimaker cura

### PR DESCRIPTION
**Pull request by jgrilo-blocks**
First definitions of Blocks machines for Ultimaker Cura

### CODE REVIEW

**Comment by Ghostkeeper** [comment]
Belongs to https://github.com/Ultimaker/fdm_materials/pull/195. They should be merged at the same time.

The tests are currently failing. It fails for a lot of files, but it seems that at least for some of them it's because the SettingVersion is outdated (should be 19 instead of 17). That indicates that you might have tested this in Cura 4.13 instead of in Cura 5.0. Do they work with Arachne?

**Comment by jgrilo-blocks** [comment]
Thanks for the feedback @Ghostkeeper, yeah i tested them on Cura 4.13 but they work in Arachne, i will fix the SettingVersion of the files

**Comment by Ghostkeeper** [comment]
You can just update the master branch of BlocksTechnology's fork with new commits. Github will automatically update this Pull Request with your latest changes. That's also the easiest for us.

**Comment by braga-dev** [comment]
@Ghostkeeper any updates when this branch will be merged?

File: resources/quality/blocks/Nozzle/blocks_global_0.20_standard.inst.cfg
Comment (nallath) [inline resources/quality/blocks/Nozzle/blocks_global_0.20_standard.inst.cfg:17]: This should not be set in a quality profile. It's already being set by the extruder. THese can (and should) be removed from all of them.

File: resources/quality/blocks/Nozzle/blocks_global_0.20_standard.inst.cfg
Comment (nallath) [inline resources/quality/blocks/Nozzle/blocks_global_0.20_standard.inst.cfg:7]: setting version is 20 now (not a big deal if you don't fix it, but it will make the tests fail right now)

File: resources/definitions/blocks_pros30.def.json
Comment (nallath) [inline resources/definitions/blocks_pros30.def.json:15]:
```suggestion
```
Remove unneeded whitespaces

File: resources/definitions/blocks_base.def.json
Comment (nallath) [inline resources/definitions/blocks_base.def.json:141]:
```suggestion
```
This is already the default in fdmprinter, no need to repeat yourself.

File: resources/definitions/blocks_base.def.json
Comment (nallath) [inline resources/definitions/blocks_base.def.json:112]:
```suggestion

```
Already the default, no need to repeat yourself

File: resources/definitions/blocks_base.def.json
Comment (nallath) [inline resources/definitions/blocks_base.def.json:116]:
```suggestion

```

File: resources/definitions/blocks_base.def.json
Comment (nallath) [inline resources/definitions/blocks_base.def.json:120]:
```suggestion

```

File: resources/definitions/blocks_base.def.json
Comment (nallath) [inline resources/definitions/blocks_base.def.json:90]:
```suggestion

```
The default material print temperature is 100 by default.

File: resources/definitions/blocks_base.def.json
Comment (nallath) [inline resources/definitions/blocks_base.def.json:48]:
```suggestion

```
The default value uses a more robust variant that is effectively the same.

File: resources/definitions/blocks_base.def.json
Comment (nallath) [inline resources/definitions/blocks_base.def.json:55]:
```suggestion

```
Default values, no need to repeat yourself.

File: resources/definitions/blocks_base.def.json
Comment (nallath) [inline resources/definitions/blocks_base.def.json:58]:
```suggestion
```
Default value

File: resources/definitions/blocks_base.def.json
Comment (nallath) [inline resources/definitions/blocks_base.def.json:60]:
```suggestion

```

File: resources/definitions/blocks_base.def.json
Comment (nallath) [inline resources/definitions/blocks_base.def.json:63]:
```suggestion

```

File: resources/definitions/blocks_base.def.json
Comment (nallath) [inline resources/definitions/blocks_base.def.json:80]:
```suggestion

```

File: resources/definitions/blocks_pros30.def.json
Comment (nallath) [inline resources/definitions/blocks_pros30.def.json:25]: Why not just define the maximum value_warning in the base file? THey seem to be the same for all machines.

File: resources/definitions/blocks_pros30.def.json
Comment (nallath) [inline resources/definitions/blocks_pros30.def.json:27]: It's the same for all printers. So why not just put it in the base file?

**Review by nallath — DISMISSED** [review]
Apologies for the slow review! I'm trying to get through all the open PR's again.

I've not checked all the settings, but I've found quite a few settings that re-define defaults. This makes it a lot harder to maintain the profile, so it's generally best to adhere to the DRY (Don't repeat yourself) as much as possible.

Furthermore, we also require buildplate meshes to be well below 1mb. Try to exclude any unneeded details like springs / screws.

**Comment by braga-dev** [comment]
@nallath changes were made and machines added. Thank you so much for the help

**Comment by braga-dev** [comment]
@nallath any chance to review the printer and quality profiles? thank you

**Comment by jellespijker** [comment]
Can you sync your fork, with our `main`. We have added a **printer-linter** tool for our printer definitions and profiles. This should help us with processing our PR's. It performs some standard diagnostic checks and apply auto-formatting on the changed files in this PR, but you can also run it locally. See https://github.com/Ultimaker/Cura/tree/main/printer-linter

If it finds issues for which it knows a fix it should post a comment with that suggested fix such that you can easily commit that fix if you're okay with it.

**Comment by braga-dev** [comment]
hello @jellespijker just sync the fork and waiting for the confirmation. thank you

**Comment by braga-dev** [comment]
@jellespijker how are you? is it possible to approve the workflow to check if there is more diagnostics with printer-liner? thank you

**Comment by braga-dev** [comment]
@jellespijker just updated with the last commits! Let's hope its the final one :D

**Review by jellespijker — CHANGES_REQUESTED** [review]
Still multiple issues and failing Unit Tests so I can't press merge :-( 

To help you guys out a bit I made a PR against your fork. See https://github.com/BlocksTechnology/Cura/pull/1

If you agree with these suggested changes feel free to merge it on your fork, this should also update this PR.

I will convert this PR to a WIP PR. You can convert it back once it is ready to be reviewed again.

**Comment by jellespijker** [comment]
@braga-dev and @jgrilo-blocks Do you guys need additional help with this PR? Or is merging my PR on your fork enough? https://github.com/BlocksTechnology/Cura/pull/1

It would be nice if we can add your printers in Cura 5.3.0

**Comment by braga-dev** [comment]
hello @jellespijker sorry for the delay. We just fixed the last issue so its all probably fine. We already tested these print profiles in Cura 5 and they are working fine so there won't be a problem. 

For future updates, its going to be easier/faster than the first ones right?

**Comment by jellespijker** [comment]
Don't worry. The main delay was originally on our side. We are finally catching up.

Yes, updates and new printers should now be processed way faster due to our improved workflow.

I will take a look at the fixes first thing o Monday. Good to hear that they have been tested on Cura 5.x.

If all goes well, these printers should be available in Cura 5.3

**Comment by jellespijker** [comment]
just mark it ready for review when you're finished so we know that we can review it.

**Comment by braga-dev** [comment]
Hello @jellespijker perfect. Great to hear that.

File: resources/definitions/blocks_base.def.json
Comment (jellespijker) [inline resources/definitions/blocks_base.def.json:67]:
```suggestion
            "default_value": 20,
```

File: resources/definitions/blocks_base.def.json
Comment (jellespijker) [inline resources/definitions/blocks_base.def.json:77]:
```suggestion
            "default_value": 40,
```

File: resources/definitions/blocks_base.def.json
Comment (jellespijker) [reply resources/definitions/blocks_base.def.json:67]: I will change these myself after I merge them

File: resources/definitions/blocks_base.def.json
Comment (jellespijker) [reply resources/definitions/blocks_base.def.json:77]: I will change these myself after I merge them

**Review by jellespijker — APPROVED** [review]
LGTM;

Two minor fixes which I will quickly fix myself when I merge it.
Congrats guys, your printers should be available in Cura 5.3

**Comment by braga-dev** [comment]
Hello @jellespijker ! Thank you very much! Can't wait.

**Pull request merged**
